src/compute/src/compute_state.rs

```diff
@@ -344,49 +417,98 @@ impl<'a, A: Allocate + 'static> ActiveComputeState<'a, A> {
     }
 
     #[tracing::instrument(level = "debug", skip(self))]
-    fn handle_peek(&mut self, peek: Peek) {
```

Comment:
I'd like this to be two separate functions instead of two large `if`/`else` blocks within the same function.